### PR TITLE
AB#530 cli doc

### PR DESCRIPTION
**Pull request by thomasten**

### CODE REVIEW

File: cmd/ego/main.go [hunk 1]
```diff
@@ -32,16 +32,6 @@ func main() {
 		}
 		sign(filename)
```
[the remark this answers is not in the document]
Comment (thomasten) [reply cmd/ego/main.go:33]: Yep, will do this together with some refactoring in another PR.

File: doc/ego_cli.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+```
+to build a Go project that uses a Makefile.
+
+## Enclave configuration file
+An enclave configuration is defined in JSON and applied when signing an executable.
+
+Here is an example configuration:
+```json
+{
+ "exe": "helloworld",
```
[the remark this answers is not in the document]
Comment (thomasten) [reply doc/ego_cli.md:105]: for me, exec means "execute" and exe means "executable"

File: doc/ego_cli.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+```
+
+`exe` is the (relative or absolute) path to the executable that should be signed.
+
+`key` is the path to the private RSA key of the signer. When invoking `ego sign` and the key file does not exist, a key with the required parameters is automatically generated. You can also generate it yourself with:
+```
+openssl genrsa -out private.pem -3 3072
+```
+
+If `debug` is true, the enclave will be debuggable.
```
[the remark this answers is not in the document]
Comment (thomasten) [reply doc/ego_cli.md:121]: What do you mean by *not enforce the other SGX parameters*?

File: doc/ego_cli.md [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+## env
+Usage:
+```
+ego env ...
+```
+Run a command within the ego build environment. For example, run
+```
+ego env make
+```
+to build a Go project that uses a Makefile.
```
[the remark this answers is not in the document]
Comment (thomasten) [reply doc/ego_cli.md:97]: This sentence begins before the command: `For example, run ... to build ...`

File: doc/ego_cli.md [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+## run
+Usage:
+```
+ego run <executable> [args...]
+```
+Run a signed executable in an enclave. You can pass arbitrary arguments to the enclave.
+
+Environment variables are only readable from within the enclave if they start with "EDG_".
+
+Set OE_SIMULATION=1 to run in simulation mode, e.g.:
```
[the remark this answers is not in the document]
Comment (thomasten) [reply doc/ego_cli.md]: like 0f20629?

File: doc/ego_cli.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (Nirusu) [reply doc/ego_cli.md:121]: Oh wait, I think I got confused here with the recent change in Marblerun where we do not enforce a sane combination of UniqueID + (SignerID || ProductID || SecurityVersion). 

There's no such thing here in ego, as far as I can see...? If so, never mind :)

File: doc/ego_cli.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (Nirusu) [reply doc/ego_cli.md:97]: Missed that, that's fine than :)

File: doc/ego_cli.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (Nirusu) [reply doc/ego_cli.md]: For development _on non-SGX enabled machines_ maybe? But looks better already.

File: doc/ego_cli.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Nirusu) [reply doc/ego_cli.md:105]: Well, we tell ego with the enclave.json to _execute_ the executable, so I do not think that's really confusing :)

I mean, there might be other better options here, but I think that "exe" just looks off in a Linux context, and I cannot remember I've ever seen a JSON config that specifies an executable as "exe" under a Linux environment. But that might just be me here, or maybe there are actually cases where this common, feel free to prove me wrong here.

Other alternatives could be 'bin', 'binary', 'program' (VS Code uses that one for their launch.json), don't think it matters that much. Just 'exe' looks kind of unusal to me though.

**Pull request merged**